src/common/snippets/src/pass/propagate_precision.cpp

```diff
@@ -0,0 +1,277 @@
+        //   2) Type relaxed based operations. Will be resolved by snippet opset.
+
+        auto input_precisions_were_changed = false;
+
+        for (const auto& input : op->inputs()) {
+            const auto convert = ngraph::as_type<snippets::op::ConvertSaturation>(input.get_source_output().get_node());
+            if (convert == nullptr) {
+                continue;
+            }
+
+            const auto precision_before = convert->input(0).get_source_output().get_element_type();
+            const auto precision_after = convert->output(0).get_element_type();
+            if (can_be_removed(precision_before, precision_after, precision_before)) {
+                op->set_argument(input.get_index(), convert->input(0).get_source_output());
+                input_precisions_were_changed = true;
+            }
+        }
+
+        std::vector<element::Type> input_precisions;
+        for (const auto& input : op->inputs()) {
+            const auto input_precision = input.get_source_output().get_element_type();
+            input_precisions.push_back(input_precision);
+        }
+
+        assert(std::all_of(
+            supported_precisions.begin(),
+            supported_precisions.end(),
+            [&input_precisions](const std::vector<element::Type>& precisions) {
+                return precisions.size() == input_precisions.size();
+            }) && "input precisions count is not equal for supported precisions");
+
+        // update input precisions
+        // if possible then convert precisions to supported
+        if (!supported_precisions.empty() &&
+            !std::any_of(
+                supported_precisions.begin(),
+                supported_precisions.end(),
+                [&input_precisions](const std::vector<element::Type>& precisions) {
+                    return precisions == input_precisions;
+                })) {
+            auto precisions = get_precisions(input_precisions,
+                                             supported_precisions);
+            if (precisions.empty()) {
+                throw ov::Exception(
+                    "there are no supported precisions for operation '" +
+                    std::string(type_info.version_id) + "::" +
+                    std::string(type_info.name) + "'");
+            }
+
+            auto find_convert = [](
+                const ngraph::Output<ngraph::Node> parent_output,
+                const ngraph::element::Type convert_type) -> snippets::op::ConvertSaturation* {
+                for (const auto& input : parent_output.get_target_inputs()) {
+                    const auto child = ngraph::as_type<snippets::op::ConvertSaturation>(input.get_node());
+                    if ((child != nullptr) && (child->output(0).get_element_type() == convert_type)) {
+                        return child;
+                    }
+                }
+                return nullptr;
+            };
+
+            for (auto i = 0ull; i < op->get_input_size(); ++i) {
```

Comment:
Minor:
```suggestion
            for (auto i = 0ul; i < op->get_input_size(); ++i) {
```

Reply:
`size_t` depeneds on platform. But on Windows (Linux?) x64 `size_t` is `unsigned long long`. `i` should be not less then `size_t`, not less then `unsigned long long`. 
Anyway, it was changed to `size_t` : )